File: Working with Excel reference tables/[C#]-Working with Excel reference tables/C#/ExcelHelper/ExcelReferenceTable.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ExcelHelper
{
    public class ExcelReferenceTable
    {
        public string Name { get; set; }
        public string SheetName { get; set; }
        public string Address { get; set; }
        public string SelectString
        {
            get
            {
                return "SELECT * FROM [" + SheetName + "$" + Address + "]";
            }
        }
        public string SourceDataFile { get; set; }
        [System.Diagnostics.DebuggerStepThrough()]
        public ExcelReferenceTable()
        {
        }
        public override string ToString()
        {
            return Name;
        }
    }
}

File: Working with Excel reference tables/[VB]-Working with Excel reference tables/VB/ExcelHelperLibrary/ExcelReferenceTable.vb
''' <summary>
''' Used to store information about data in an Excel file
''' </summary>
''' <remarks>
''' If it bothers you that read/write properties should be read-only
''' feel free to make them read-only.
''' </remarks>
Public Class ExcelReferenceTable
    Public Property Name As String
    Public Property SheetName As String
    Public Property Address As String
    Public ReadOnly Property SelectString As String
        Get
            Return "SELECT * FROM [" & SheetName & "$" & Address & "]"
        End Get
    End Property
    Public Property SourceDataFile As String
    <System.Diagnostics.DebuggerStepThrough()> _
    Public Sub New()
    End Sub
    Public Overrides Function ToString() As String
        Return Name
    End Function
End Class
